nixos/modules/services/audio/mpdscribble.nix

```
@@ -0,0 +1,236 @@
+      '';
+    };
+
+    host = mkOption {
+      default = (if mpdCfg.network.listenAddress != "any" then
+        mpdCfg.network.listenAddress
+      else
+        "localhost");
+      type = types.str;
+      description = ''
+        Host for the mpdscribble daemon to search for a mpd daemon on.
+      '';
+    };
+
+    password = mkOption {
+      default = null;
+      type = types.nullOr types.str;
+      description = ''
+        Password for the mpd daemon.
+        Attention: This will be stored in the globally readable /nix/store.
+      '';
+    };
+
+    passwordFile = mkOption {
+      default = null;
+      type = types.nullOr types.str;
+      description = ''
+        File containing the password for the mpd daemon.
+      '';
+    };
+
+    port = mkOption {
+      default = mpdCfg.network.port;
+      type = types.port;
+      description = ''
+        Port for the mpdscribble daemon to search for a mpd daemon on.
+      '';
+    };
+
+    endpoints = mkOption {
+      type = (let
+        endpoint = { name, ... }: {
+          options = {
+            url = mkOption {
+              type = types.str;
+              default = endpointUrls.${name} or "";
+              description =
+                "The url endpoint where the scrobble API is listening.";
+            };
+            username = mkOption {
+              type = types.str;
+              description = ''
+                Username for the scrobble service.
+              '';
+            };
+            password = mkOption {
+              type = types.str;
+              description = ''
+                Password, either as MD5SUM or cleartext.
+                Attention: This will be stored in the globally readable /nix/store.
+              '';
+            };
+            passwordFile = mkOption {
+              type = types.nullOr types.str;
+              description =
+                "File containing the password, either as MD5SUM or cleartext.";
+            };
+            journal = mkOption {
+              type = types.path;
+              default = "/var/lib/mpdscribble/${name}.journal";
+              description = "The path where mpdscribble will store the journal";
+            };
+          };
+        };
+      in types.attrsOf (types.submodule endpoint));
+      default = { };
+      example = {
+        "last.fm" = {
+          url = "http://post.audioscrobbler.com/";
+          username = "foo";
+          passwordFile = "/run/secrets/mpd_password";
+          journal = "/var/lib/mpdscribble/lastfm.journal";
+        };
+      };
+      description = ''
+        Endpoints to scrobble to.
+        If the endpoint is one of ${
+          concatStringsSep ", " (attrNames endpointUrls)
+        } the url is set automatically.
+      '';
+    };
+
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    systemd.services.mpdscribble = {
+      after = [ "mpd.service" "network.target" ];
```

Maybe this shouldn't be mandatory in case one uses a remote mpd?

```suggestion
      after = [ "network.target" ];
```

---

You could check if the user is connecting to `localhost` and conditionally add based on that.

---

I've added a check on "localhost", however this does not work n the case where the user is connecting to the local mpd through the public ip. Not sure how to best solve this.

---

We can't really check for that scenario, and I don't think that is a problem.